src/theme-preval.ts

```diff
@@ -374,3 +384,70 @@ export const theme = {
 function fontStack(fonts: string[]) {
   return fonts.map(font => (font.includes(' ') ? `"${font}"` : font)).join(', ')
 }
+
+// The following few methods are data wrangling to split apart shadow values from non-shadow values
+// and also ensure the resulting objects don't contain empty objects from filtering
+// These methods are a temporary measure until we push these changes upstream to primer primitives
+
+function shadowFilter(o: { [key: string]: any }, filterShadow: boolean) {
+  const newObject: { [key: string]: any } = {}
+  Object.keys(o).forEach(function(key: string) {
+    const value: any | undefined = o[key];
+    if (typeof value == "object" && !Array.isArray(value)) {
+      newObject[key] = shadowFilter(value, filterShadow)
+    } else if (typeof value === "string" && value.match(/(inset\s|)([0-9.empx\s]+){1,4}rgb[a]?\(.*\)/)) {
+      // this is a shadow value
+      if (filterShadow) {
+        newObject[key] = value
+      }
+    } else if (typeof value === "string") {
+      // this is any other non shadow string value
+      if (!filterShadow) {
+        newObject[key] = value
+      }
+    } else {
+      newObject[key] = value
+    }
+  })
+  return newObject;
+}
+
+function recursivelyFilterEmptyObjects(o: { [key: string]: any }) {
+  const newObject: { [key: string]: any } = {}
+  Object.keys(o).forEach(function(key: string) {
+    const value: any | undefined = o[key];
+    if (typeof value == "object" && !Array.isArray(value)) {
+      if (isObjectRecursivelyPresent(value)) {
+        newObject[key] = value
+      } else {
+        const filteredChild = recursivelyFilterEmptyObjects(value)
+        if (Object.keys(filteredChild).length > 0) {
+          newObject[key] = filteredChild
+        }
+      }
+    } else {
+      newObject[key] = value
+    }
+  })
+  return newObject;
+}
+
+function isObjectRecursivelyPresent(o: { [key: string]: any }) {
+  const presenceValues: boolean[] = []
+  if (Object.keys(o).length === 0) { return false }
+  Object.keys(o).forEach(function(key: string) {
+    const value: any | undefined = o[key];
+    if (typeof value === 'object' && !Array.isArray(value)) {
+      if (Object.keys(value).length === 0) {
+        presenceValues.push(false)
+      } else {
+        const isRecursivelyPresent = isObjectRecursivelyPresent(value)
+        presenceValues.push(isRecursivelyPresent)
+      }
+    } else {
+      presenceValues.push(true)
+    }
+  })
+
+  return presenceValues.every(p=>p)
+}
```

Comment:
These methods are probably pretty difficult to parse, and although they are temporary measures, I'm not sure if that amount of technical debt is okay here for the moment?

Essentially, they recursively scrub through the colors object and filter out shadows from colors, and then recursively clean those objects up to make sure there are not a bunch of empty objects in the hierarchy. This should be fairly robust to any changes in primer primitives that occur before we push upstream changes back. I put the resulting JSON for both shadows and colors into this gist:

https://gist.github.com/VanAnderson/a68f5f1fa08ae7b5aee728c0666fd90c

These objects correspond to `filteredShadows` and `filteredColors` on L355 and L356.

Reply:
Just a thought - if this code is too tech-debty, we could also just copy the resulting JSON here and keep those values on the primer/components side until we push those changes upstream to primer/primitives - not very resilient to changes in primer/primitives, but sure easier to understand!

Reply:
I agree this is hard to parse but I'm okay with it as a temporary measure, especially since it's labeled as temporary in a comment and we have plans to remove it as part of this epic.

We should probably add at least one unit test though.

Reply:
Oh, I think I figured out a simpler way to write this function. What do you think of this approach?

https://runkit.com/colebemis/60369315e30c620013489a4f